pkg/cmd/step/create/step_create_task_test.go

```diff
@@ -86,7 +86,7 @@ func TestGenerateTektonCRDs(t *testing.T) {
 		{
 			name:         "js_build_pack",
 			language:     "javascript",
-			repoName:     "js-test-repo",
+			repoName:     "js.test.repo",
```

Comment:
@mrageh so what are we really proving here? AFAICT all occurrences of '-' are replaced with '.' for this test case. However, I cannot see where ' js.test.repo' would become 'js-test-repo' which would be what we want to prove, right?

Might be that this particular test you are modifying is not using Kaniko? If we cannot test it, I'd rather not change anything in the test files. Better to know there is no testing opposed to looking at the commit in a few months time and thinking there was a test added and wondering how it works. 

@abayer - wdyt?

The other thing here is that at best a change to _step_create_task_test.go_ will test the mechanics of _modifyStep_ and that the replacement occurs. It will not assert that the actual reported issue is resolved.

Reply:
@hferentschik So the thing that's relevant here is what _hasn't_ changed: https://github.com/jenkins-x/jx/pull/6789/files#diff-6f8d554e2e7907cb7ea6450cf8c34aa0R288 =)

Reply:
@hferentschik check this file out https://github.com/jenkins-x/jx/blob/456b4f42baa6dae0dc539d54a500a7878d54fd01/pkg/cmd/step/create/test_data/step_create_task/js_build_pack/tasks.yml#L288, as you can see the docker registry name was successfully converted to `js-test-repo`. Which is what this PR fixes, the test is very obscure I know. ¯\_(ツ)_/¯

Reply:
Thanks, that's what I was looking for. Since it had not changed I did not see it in the diff. 

> Which is what this PR fixes, the test is very obscure I know

Got it now. Very subtle indeed.